lib/events/stream_test.go

```diff
@@ -370,6 +372,96 @@ func TestEncryptedRecordingIO(t *testing.T) {
 	require.Len(t, decryptedEvents, eventCount+2)
 }
 
+func TestSummarization(t *testing.T) {
+	cases := []struct {
+		name               string
+		useSummarizer      bool
+		summarizationError error
+	}{
+		{
+			name:          "noop summarizer",
+			useSummarizer: false,
+		},
+		{
+			name:          "successful summarization",
+			useSummarizer: true,
+		},
+		{
+			// Since the behavior differs only in logging, this test is here just to
+			// make sure an error doesn't cause a panic.
+			name:               "summarization error",
+			useSummarizer:      true,
+			summarizationError: errors.New("summarization error"),
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ctx, cancel := context.WithTimeout(t.Context(), 10*time.Second)
+			defer cancel()
```

Comment:
If your tests don't actually depend on the 10-second timeout, you can just use `t.Context()` for a context.